nemo/nemo/utils/exp_logging.py

```diff
@@ -80,20 +83,43 @@ def __init__(
         # tb_dir behaves the same as ckpt_dir except default folder name is
         # tensorboard instead of checkpoints
         self.tb_dir = tb_dir
+        tm_suf = time.strftime('%Y-%m-%d_%H-%M-%S')
+        if local_rank is not None and add_time:
+            if broadcast_func is None:
+                raise ValueError(
+                    "local rank was not None, but ExpManager was not passed a "
+                    "broadcast function to broadcast the datetime suffix")
+            if local_rank == 0:
+                broadcast_func(string=tm_suf)
+            else:
+                tm_suf = broadcast_func(str_len=len(tm_suf))
 
         # Create work_dir if specified
         if work_dir:
             self.work_dir = work_dir
             if add_time:
-                self.work_dir = os.path.join(
-                    work_dir, time.strftime('%Y%m%d-%H%M%S'))
+                self.work_dir = os.path.join(work_dir, tm_suf)
             self.make_dir(self.work_dir, exist_ok)
             if use_tb:
                 self.get_tb_writer(exist_ok=exist_ok)
             self.ckpt_dir = f'{self.work_dir}/checkpoints'
-            if files_to_copy:
+            if files_to_copy and self.local_rank == 0:
                 for file in files_to_copy:
-                    copy_wo_overwrite(self.work_dir, file)
+                    copy_wo_overwrite(self.work_dir, file, tm_suf)
+            if self.local_rank == 0:
```

Comment:
and what if it is None? No output dir?

Reply:
It is set to 0 if None on the first line of init: `self.local_rank = local_rank if local_rank is not None else 0`

Reply:
So 0 should be default value, right?